src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -943,19 +961,23 @@ class IMain(@BeanProperty val factory: ScriptEngineFactory, initialSettings: Set
        */
       val evalResult = Request.this.value match {
         case NoSymbol => ""
-        case sym      => "lazy val %s = %s".format(lineRep.resultName, originalPath(sym))
+        case sym      => "lazy val %s = { compute ; %s }".format(lineRep.resultName, originalPath(sym))
       }
       // first line evaluates object to make sure constructor is run
+      val initLine = settings.YreplWrap.value match {
+        case "app" => s"($fullAccessPath main null)"
+        case _     => fullAccessPath
+      }
       // initial "" so later code can uniformly be: + etc
```

Comment:
I'll move to 2.12. I'm not sure the init line matters, but it does strip a layer of wrapping. Maybe somebody is counting the $iw's somewhere. Nobody's asking for the feature, and by 2.12 everyone will be using worksheet or ammonite, so it really won't matter.